dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskInstanceServiceImpl.java

```diff
@@ -306,4 +306,31 @@ public Result stopTask(User loginUser, long projectCode, Integer taskInstanceId)
 
         return result;
     }
+
+    @Override
+    public Result queryTaskInstanceByCode(User loginUser, long projectCode, Long taskCode) {
```

Comment:
This is a new method, I think we should return a exact object but not result, WDYT? @zhongjiajie 
```suggestion
    public TaskInstance queryTaskInstanceByCode(User loginUser, long projectCode, Long taskCode) {
```

Reply:
Yeah, we should return the entity instead of  Result object, the pros is when we have cli or API, they could use the return entity directly instead of by parsing the result object
I think the results object should be wrapper by controller.